qst/src/main/java/io/deephaven/qst/type/Type.java

```diff
@@ -12,19 +13,29 @@
 public interface Type<T> {
 
     /**
-     * Finds the {@link #knownTypes() known type}, or else creates a {@link CustomType custom type}.
+     * Finds the {@link #knownTypes() known type}, or else creates the relevant
+     * {@link NativeArrayType native array type} or {@link CustomType custom type}.
      *
      * @param clazz the class
      * @param <T> the generic type of {@code clazz}
      * @return the type
      */
     static <T> Type<T> find(Class<T> clazz) {
-        return TypeHelper.findStatic(clazz).orElseGet(() -> CustomType.of(clazz));
+        Optional<Type<T>> found = TypeHelper.findStatic(clazz);
+        if (found.isPresent()) {
+            return found.get();
+        }
+        if (clazz.isArray()) {
+            return NativeArrayType.of(clazz, find(clazz.getComponentType()));
+        }
+        return found.orElseGet(() -> CustomType.of(clazz));
     }
 
     /**
-     * The list of known types. Includes the universe of all {@link PrimitiveType primitive types}
-     * and {@link GenericType generic types}, except for {@link CustomType custom types}.
+     * The list of known types. Includes the universe of {@link PrimitiveType primitive types} and
```

Comment:
There's a stray period, and my brain can't parse the existing formulation, please revisit.